src/codemodder/codemodder.py

```diff
@@ -156,22 +156,13 @@ def run(original_args) -> int:
     logger.info("codemodder: python/%s", __version__)
     logger.info("command: %s %s", Path(sys.argv[0]).name, " ".join(original_args))
 
-    # TODO: sonar files should be _parsed_ here as well
     # TODO: this should be dict[str, list[Path]]
-
     tool_result_files_map: DefaultDict[str, list[str]] = detect_sarif_tools(
         [Path(name) for name in argv.sarif or []]
     )
-    (
-        tool_result_files_map["sonar"].extend(argv.sonar_issues_json)
-        if argv.sonar_issues_json
-        else None
-    )
-    (
-        tool_result_files_map["sonar"].extend(argv.sonar_hotspots_json)
-        if argv.sonar_hotspots_json
-        else None
-    )
+    tool_result_files_map["sonar"].extend(argv.sonar_issues_json or [])
+    tool_result_files_map["sonar"].extend(argv.sonar_hotspots_json or [])
+    tool_result_files_map["defectdojo"] = argv.defectdojo_findings_json or []
```

Comment:
I changed this logic in [this PR](https://github.com/pixee/codemodder-python/pull/433/files) in favor of the if/else None because as you have it now, `tool_result_files_map` will have a `sonar` key even if we don't have any sonar findings, I don't think that's what we want? with if/else None, nothing is added if the argv. is empty

Reply:
Each of the tool detectors does something like this:
```python
context.tool_result_files_map.get("sonar", [])
```

Also using `None` defeats the purpose of having a `defaultdict` here. I think it's cleaner just to set it to an empty array.

Reply:
The if/else None isn't doing this:
`defaultdict(<class 'list'>, {'sonar': None})`
it's doing this:
`defaultdict(<class 'list'>, {})` # totally empty

which is what I think we want if sonar json isn't used.

Reply:
I see. I don't think it's a problem to have the sonar key since in either case the detector fills the result with an empty list.